test/endpoints/baz/baz_metrics_test.go

```diff
@@ -146,7 +147,77 @@ func TestCallMetrics(t *testing.T) {
 		assert.Equal(t, expectedEndpoitTags[tag.GetTagName()], tag.GetTagValue())
 	}
 
-	tchannelOutboundSuccessMetric := metrics[3]
+	expectedJaegerSpanMetricName := "test-gateway.production.all-workers.jaeger.spans"
```

Comment:
We can use `MockAgent` which will run a fake agent on a UDP port to be able to write end-to-end tests.

https://github.com/uber/jaeger-client-go/blob/master/testutils/mock_agent.go#L43-L65

It has a `agent.SpanServerAddr()` method so we can configure the address of the reporter with `ReporterConfig.LocalAgentHostPort`